src/supervisor/watchers/handlers/argo-rollout.ts

```diff
@@ -0,0 +1,215 @@
+
+  await deleteWorkload(
+    {
+      kind: WorkloadKind.ArgoRollout,
+      objectMeta: rollout.metadata,
+      specMeta: rollout.spec.template.metadata,
+      ownerRefs: rollout.metadata.ownerReferences,
+      revision: rollout.status.observedGeneration,
+      podSpec: rollout.spec.template.spec,
+    },
+    workloadName,
+  );
+}
+
+export async function isNamespacedArgoRolloutSupported(
+  namespace: string,
+): Promise<boolean> {
+  try {
+    const pretty = undefined;
+    const continueToken = undefined;
+    const fieldSelector = undefined;
+    const labelSelector = undefined;
+    const limit = 1; // Try to grab only a single object
+    const resourceVersion = undefined; // List anything in the cluster
+    const timeoutSeconds = 10; // Don't block the snyk-monitor indefinitely
+    const attemptedApiCall = await retryKubernetesApiRequest(() =>
+      k8sApi.customObjectsClient.listNamespacedCustomObject(
+        'argoproj.io',
+        'v1alpha1',
+        namespace,
+        'rollouts',
+        pretty,
+        continueToken,
+        fieldSelector,
+        labelSelector,
+        limit,
+        resourceVersion,
+        timeoutSeconds,
+      ),
+    );
+    return (
+      attemptedApiCall !== undefined &&
+      attemptedApiCall.response !== undefined &&
+      attemptedApiCall.response.statusCode !== undefined &&
+      attemptedApiCall.response.statusCode >= 200 &&
+      attemptedApiCall.response.statusCode < 300
+    );
+  } catch (error) {
+    logger.debug(
+      { error, workloadKind: WorkloadKind.ArgoRollout },
+      'Failed on Kubernetes API call to list namespaced argoproj.io/Rollout',
+    );
+    return false;
+  }
+}
+
+export async function isClusterArgoRolloutSupported(): Promise<boolean> {
+  try {
+    const pretty = undefined;
+    const continueToken = undefined;
+    const fieldSelector = undefined;
+    const labelSelector = undefined;
+    const limit = 1; // Try to grab only a single object
+    const resourceVersion = undefined; // List anything in the cluster
+    const timeoutSeconds = 10; // Don't block the snyk-monitor indefinitely
+    const attemptedApiCall = await retryKubernetesApiRequest(() =>
+      k8sApi.customObjectsClient.listClusterCustomObject(
+        'argoproj.io',
+        'v1alpha1',
+        'rollouts',
+        pretty,
+        continueToken,
+        fieldSelector,
+        labelSelector,
+        limit,
+        resourceVersion,
+        timeoutSeconds,
+      ),
+    );
+    return (
+      attemptedApiCall !== undefined &&
+      attemptedApiCall.response !== undefined &&
+      attemptedApiCall.response.statusCode !== undefined &&
+      attemptedApiCall.response.statusCode >= 200 &&
+      attemptedApiCall.response.statusCode < 300
+    );
+  } catch (error) {
+    logger.debug(
+      { error, workloadKind: WorkloadKind.ArgoRollout },
+      'Failed on Kubernetes API call to list cluster argoproj.io/Rollout',
+    );
+    return false;
```

Comment:
should we log somewhere that it is or isn't supported?

Reply:
Yes it happens one level above - it logs that the workload type is not supported

Reply:
ah yes, I didn't go far enough back to see the log :)